dashboards-observability/public/components/custom_panels/custom_panel_view.tsx

```diff
@@ -346,6 +348,29 @@ export const CustomPanelView = ({
     }
   };
 
+  const buildBaseQuery = async () => {
+    const indices: string[] = [];
+    for (let i = 0; i < panelVisualizations.length; i++) {
+      const visualizationId = panelVisualizations[i].savedVisualizationId;
+      await http
+        .get(`${CUSTOM_PANELS_API_PREFIX}/visualizations/${visualizationId}`)
```

Comment:
maybe todo, you can get objects by a list of ids in one http request instead of sending requests in a loop

Reply:
Yes, definitely a todo